core/src/test/java/org/javarosa/core/model/test/FormDefTest.java

```diff
@@ -432,7 +440,57 @@ public void testSimilarBindConditionsAreDistinguished() throws Exception {
         }
     }
 
-
+    // regression test for when we weren't decrementing multiplicities correctly when repeats were deleted
+    @Test
+    public void testDeleteRepeatMultiplicities() throws IOException {
+        FormParseInit fpi = new FormParseInit("/multiple_repeats.xml");
+        FormEntryController fec = initFormEntry(fpi, "en");
+        fec.stepToNextEvent();
+        fec.newRepeat();
+        fec.stepToNextEvent();
+        fec.answerQuestion(new StringData("First repeat, first iteration: question2"));
+        fec.stepToNextEvent();
+        fec.answerQuestion(new StringData("First repeat, first iteration: question3"));
+        fec.stepToNextEvent();
+        fec.newRepeat();
+        fec.stepToNextEvent();
+        fec.answerQuestion(new StringData("First repeat, second iteration: question2"));
+        fec.stepToNextEvent();
+        fec.answerQuestion(new StringData("First repeat, second iteration: question3"));
+        fec.stepToNextEvent();
+        fec.stepToNextEvent();
+        // moving from first repeat (question1) to second (question4)
+        fec.newRepeat();
+        fec.stepToNextEvent();
+        fec.answerQuestion(new StringData("Second repeat, first iteration: question5"));
+        fec.stepToNextEvent();
+        fec.answerQuestion(new StringData("Second repeat, first iteration: question6"));
+        fec.stepToNextEvent();
+        fec.newRepeat();
+        fec.stepToNextEvent();
+        fec.answerQuestion(new StringData("Second repeat, second iteration: question5"));
+        fec.stepToNextEvent();
+        fec.answerQuestion(new StringData("Second repeat, second iteration: question6"));
+        fec.stepToPreviousEvent();
+        fec.stepToPreviousEvent();
+
+        TreeElement root = fpi.getFormDef().getInstance().getRoot();
+
+        // confirm both groups have two iterations and second iteration is set
+        assertTrue(root.getChildMultiplicity("question4") == 2);
+        assertTrue(root.getChild("question4", 1) != null);
+        assertTrue(root.getChildMultiplicity("question1") == 20;
+        assertTrue(root.getChild("question1", 1) != null);
+
+        fec.deleteRepeat(0);
+
+        // Confirm that the deleted repeat is gone and its sibling's multiplicity reduced
+        assertTrue(root.getChildMultiplicity("question4") == 1);
+        assertTrue(root.getChild("question4", 1) == null);
+        // Confirm that the other repeat is unchanged
+        assertTrue(root.getChildMultiplicity("question1") == 2);
+        assertTrue(root.getChild("question1", 1) != null);
```

Comment:
Last nitpick: instead of doing `assertTrue(a != b)` and `assertTrue(a == b) can you do `assertNotEquals(a, b)` and `assertEquals(a, b)`

When `b` is `null`: `assertNotNull(a)`